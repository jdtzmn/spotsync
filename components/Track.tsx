import React, { Component } from 'react'
import { runIfIsFn } from './helpers'
import css from '../styles/components/track.scss'

interface TrackProps {
  name: string,
  cover: string,
  artists: string[],
  altText?: string
  onAltClick?: (event) => void
}

class Track extends Component<TrackProps> {
  constructor(props) {
    super(props)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick () {
    const { onAltClick } = this.props
    runIfIsFn(onAltClick)
  }

  render() {
    const { name, cover, artists, altText } = this.props
    const artistsText = artists.join(', ')

    return (
      <div className={css.track}>
        <img
          src={cover}
          className={css.cover}
          alt=""
        />
        <div className={css.detailsContainer}>
          <div className={css.details}>
            <h1 className={css.name}>{name}</h1>
            <sub className={css.artists}>{artistsText}</sub>
          </div>
        </div>
        <div className={css.verticalAlign}>
          {altText ? <div className={css.altText} onClick={this.handleClick}>{altText}</div> : undefined}
        </div>
      </div>
    )
  }
}

export default Track
