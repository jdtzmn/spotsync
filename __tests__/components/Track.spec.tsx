import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Track from 'components/Track'

describe('ResponsiveParty', () => {
  let wrapper: ShallowWrapper
  beforeEach(() => {
    wrapper = shallow(<Track name='' cover='' artists={[]} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render based on props', () => {
    // set props
    const props = {
      name: 'Test Name',
      cover: 'https://cover.url',
      artists: ['Artist 1', 'Artist 2'],
      altText: 'Test altText'
    }
    wrapper.setProps(props)

    // make sure it renders correctly
    expect(wrapper
      .find('img')
      .prop('src'))
      .toBe(props.cover)
    expect(wrapper
      .find('h1')
      .text())
      .toBe(props.name)
    expect(wrapper
      .find('sub')
      .text())
      .toBe(props.artists.join(', '))
    expect(wrapper
      .find('div')
      .last()
      .text())
      .toBe(props.altText)
  })

  it('should call onAltClick when altText is clicked', () => {
    // setup props
    const onAltClick = jest.fn()
    wrapper.setProps({
      altText: 'click me',
      onAltClick
    })
    wrapper.instance().forceUpdate()

    // click the altText
    wrapper
      .find('div')
      .last()
      .simulate('click')

    expect(onAltClick).toHaveBeenCalled()
  })
})
