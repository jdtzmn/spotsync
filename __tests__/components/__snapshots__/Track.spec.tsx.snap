// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveParty should render correctly 1`] = `
<div>
  <img
    alt=""
    src=""
  />
  <div>
    <div>
      <h1 />
      <sub />
    </div>
  </div>
  <div />
</div>
`;
